### PR TITLE
Release 0 0 6

### PR DESCRIPTION
**Pull request by eirrgang**
gmxpi 0.0.6 release

Interface and feature updates

- Updates to :mod:`gmx.version` module
- Automatically set and restore from MD simulation checkpoints in the session working directory.
- Allow control of whether simulation output is appended or truncated (PR #126).
- Allow plugins to issue a stop signal to MD simulations (reference #62 for gromacs-gmxapi and sample_restraint repos).
- Changes to `gmx.exceptions`
- Allow full CMake-driven install
- Updated example notebooks in sample_restraint repository.

Internal

- Improved CI testing
- #64 Unique work spec identification.

Bug fixes

- #66 Docker does not access current gmxpy version.
- #123 Race condition in session closing.

Remaining tasks

- [x] README.md should have citation information for gmxapi (bioinformatics 2018).
- [x] check for embedded "to do"s
- [x] remove shared_data_maker and test code, discussion, to one or more of #34, #36, #37, #45, #69
- [x] reword gmx.Status description, and cite appropriate tracked issue or design document.
- [x] reconsider comment lines used to hint text-block separation in workflow.py docstring.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -210,7 +210,7 @@ def build_extension(self, ext):
         if build_gromacs:
             # TODO! We need to distinguish dev branch builds from master branch builds or always build with the
             # master branch of the dependency. For one thing, as is, this line needs to be toggled for every release.
-            gromacs_url = "https://github.com/kassonlab/gromacs-gmxapi/archive/dev_5.zip"
+            gromacs_url = "https://github.com/kassonlab/gromacs-gmxapi/archive/devel.zip"
```
Comment (peterkasson) [inline setup.py]: Is there a compelling reason to use a static zip file for gromacs_url rather than a git branch (or tag) checkout?

File: src/gmx/context.py [hunk 2]
```diff
@@ -157,91 +157,92 @@ def __init__(self, work):
...
-            # but we aren't there yet.
-            runner = None
-            return runner
-
-    builder = Builder(element)
-    return builder
+# def shared_data_maker(element):
+#     """Make a shared data element for use by dependent nodes.
+#
+#     Note: outdated. This code is in limbo and should not be used.
```
Comment (peterkasson) [inline src/gmx/context.py]: What is the meaning of "in limbo" here?  Deprecated?  In the process of being refactored?

File: src/gmx/status.py [hunk 3]
```diff
@@ -1,7 +1,9 @@
 class Status(object):
     """Ultimately hold status for API operation.
 
-    This should probably inherit from gmx.core.Status.
+    Either this should be inherited by gmx.core.Status or a reference to it or it gets
```
Comment (peterkasson) [inline src/gmx/status.py]: syntax of this comment is a little hard to parse.

File: src/gmx/workflow.py [hunk 4]
```diff
@@ -36,6 +17,8 @@
     >>> with gmx.context.Context(md.workspec) as session:
     ...    session.run()
 
+.. This comment line seems to be necessary to help parsing in some cases.
```
Comment (peterkasson) [inline src/gmx/workflow.py]: ??

File: setup.py (hunk 1 above)
Comment (eirrgang) [reply setup.py]: It is much faster than cloning the repo and uses standard Python modules. More recently, I've found the fastest thing is `git clone -depth 1 ...`, but when we move off of readthedocs.org to a push from Travis-CI, we will not need to work so hard to shave seconds off the build time.

File: src/gmx/context.py (hunk 2 above)
Comment (eirrgang) [reply src/gmx/context.py]: Ah. Thanks. I should have tracked that better. There is a lot of code in the various repos that is currently unused or completely dead. I'm in the process of removing the dead code where it is untested, but for unused code that is tested, I have been content to make minimal changes. For this particular section, I think the code was tested and error free, but unused and inconsistent with forward-looking design. I'll take a closer look at it before letting it propagate further.

To do

- [ ] make better use of "to do"s
- [ ] remove shared_data_maker and test code, discussion, to one or more of #34, #36, #37, #45, #69

Comment (eirrgang) [reply src/gmx/context.py]: There's not really a good place to stash ideas for posterity that I can find with GitHub. Unless maybe that's what Gist is for? I would want to stash the commented out bits of context.py and of test_mpiarraycontext.py. As it is, I tried to be more consistent with the commented out code and to include comments referring to tracked issues with commit 848e5c1. If you have other ideas, please let me know.

**Comment by eirrgang** [comment]
The conversation referenced by @peterkasson relates to the use of a GitHub release archive URL (zip file) rather than git based source retrieval. The response was

> It is much faster than cloning the repo and uses standard Python modules. More recently, I've found the fastest thing is git clone -depth 1 ..., but when we move off of readthedocs.org to a push from Travis-CI, we will not need to work so hard to shave seconds off the build time.

I will add a comment to this effect where archive URLs are referenced in the source. But, to clarify, is there a reason to discourage use of, say, `wget` over `git`? I don't believe git is currently described as a requirement for installation, but I can check that, too, and make sure documentation is consistent with reality.

**Comment by peterkasson** [comment]
My primary concern is just that if we pull a zip file, then unless that is automatically created every time then we have the potential for the zip file and the source tree to get out of sync.  If we're pulling a designated tag from the source tree, then it's clearly defined.
